cmake/horovod/CMakeLists.txt

```diff
@@ -46,9 +40,46 @@ file(GLOB_RECURSE horovod_common_src
 
 add_library(horovod ${horovod_common_src})
 
-target_compile_options(horovod PRIVATE ${MPI_CXX_COMPILE_FLAGS})
-target_compile_options(horovod PRIVATE -march=native)
-target_compile_options(horovod PRIVATE -Wno-error=sign-compare)
-target_compile_options(horovod PRIVATE -Wno-reorder)
-target_compile_options(horovod PRIVATE -Wno-unused-variable)
-target_include_directories(horovod PRIVATE ${THIRD_PARTY_ROOT}/eigen ${THIRD_PARTY_ROOT}/lbfgs/include ${THIRD_PARTY_ROOT}/boost/assert/include ${THIRD_PARTY_ROOT}/boost/config/include ${THIRD_PARTY_ROOT}/boost/core/include ${THIRD_PARTY_ROOT}/boost/detail/include ${THIRD_PARTY_ROOT}/boost/iterator/include ${THIRD_PARTY_ROOT}/boost/lockfree/include ${THIRD_PARTY_ROOT}/boost/mpl/include ${THIRD_PARTY_ROOT}/boost/parameter/include ${THIRD_PARTY_ROOT}/boost/predef/include ${THIRD_PARTY_ROOT}/boost/preprocessor/include ${THIRD_PARTY_ROOT}/boost/static_assert/include ${THIRD_PARTY_ROOT}/boost/type_traits/include ${THIRD_PARTY_ROOT}/boost/utility/include ${THIRD_PARTY_ROOT}/flatbuffers/include /usr/local/cuda/include ${MPI_CXX_INCLUDE_PATH})
+target_compile_definitions(horovod PRIVATE
+    EIGEN_MPL2_ONLY=1
+    HAVE_CUDA=1
+    HAVE_NCCL=1
+    HAVE_MPI
+    HOROVOD_GPU_ALLREDUCE='N'
+    HOROVOD_GPU_ALLGATHER='M'
+    HOROVOD_GPU_BROADCAST='M')
+
+target_compile_options(horovod PRIVATE
+    -pthread
+    -Wl,--sysroot=/
+    -fwrapv
+    -fPIC
+    ${MPI_CXX_COMPILE_FLAGS}
+    -march=native
```

Comment:
Please remove "-march=native". Otherwise the result binary can only work on the machine built it.

Reply:
hm, i'm seeing compile errors when i do that:

```
/usr/lib/gcc/x86_64-linux-gnu/5/include/f16cintrin.h: In member function ‘__m256 horovod::common::Adasum<Communicator_type>::MmLoaduPh(const uint16_t*) [with Communicator_type = ompi_communicator_t*; __m256 = __vector(8) float; uint16_t = short unsigned int]’:
/usr/lib/gcc/x86_64-linux-gnu/5/include/f16cintrin.h:52:1: error: inlining failed in call to always_inline ‘__m256 _mm256_cvtph_ps(__m128i)’: target specific option mismatch
 _mm256_cvtph_ps (__m128i __A)
...
onnxruntime/cmake/external/horovod/horovod/common/ops/adasum/adasum.h:494:27: error: called from here
     return _mm256_cvtph_ps(r);
```

Reply:
As a temp solution, you may remove "-march=native" here and turn on onnxruntime_BUILD_FOR_NATIVE_MACHINE.  But eventually we need to solve it.

Reply:
ok. i think the plan is to remove Horovod eventually, so this should be temporary.

Reply:
Good.

Reply:
hm, i tried to set onnxruntime_BUILD_FOR_NATIVE_MACHINE, but that only adds the "-march=native" flag for release builds. will just leave it for now as it is temporary.